src/Raku/ast/expressions.rakumod

```diff
@@ -1982,21 +1982,34 @@ class RakuAST::ApplyInfix
                 }
             }
 
-            my $type := self.left.return-type;
+            my $type := $left.return-type;
             # Subset type checking can have side effects, so don't do that at compile time.
-            if nqp::istype($infix, RakuAST::Assignment) && !nqp::eqaddr($type, Mu) && !nqp::istype($type.HOW, Perl6::Metamodel::SubsetHOW) {
-                my $right := self.right;
-                if nqp::istype($right,RakuAST::Literal) {
-                    if nqp::objprimspec($type) {
-                        $type := $type.HOW.mro($type)[1];
-                    }
+            if (nqp::istype($infix, RakuAST::Assignment) || $infix.operator eq ':=')
+                && !nqp::istype($type.HOW, Perl6::Metamodel::SubsetHOW) {
 
-                    my $value := $right.compile-time-value;
-                    if !nqp::istype($value, $type)
-                      && nqp::istype($type, $resolver.type-from-setting('Numeric')) {
-                        self.add-sorry:
-                          $resolver.build-exception: 'X::Syntax::Number::LiteralType',
-                            :vartype($type), :$value;
+                my $sigil := (try $left.sigil) // '';
```

Comment:
In general I avoid using try in the RakuAST code. Reason is that it doesn't tell the reader in which situations it's expected to fail or succeed. Also it may catch exceptions that you didn't think of. Granted, that's unlikely with such a basic accessor as .sigil. But what if someone adds a method sigil to some subclass that does something completely different and fails?

Reply:
Or what if a node has a sigil method for some reason (maybe a convenience wrapper) but you did not expect this type of node to be included here?

Reply:
> In general I avoid using try in the RakuAST code. Reason is that it doesn't tell the reader in which situations it's expected to fail or succeed. Also it may catch exceptions that you didn't think of. Granted, that's unlikely with such a basic accessor as .sigil. But what if someone adds a method sigil to some subclass that does something completely different and fails?

agree, I saw three place using try in the RakuAST code, and I copied it from here: https://github.com/Rakudo/Rakudo/commit/a0ed2ae93a3a8d0fa3de5fb079c464bec5f530be#diff-3c8aafcca6a26bfab047f06f101aa7d47de2d605e66ab2c280108063bda8f291L794

Reply:
I think it would be better to have `RAKUAST::TypeCheck` and `RAKUAST::TypeCheck::Parameter` class to reuse these codes?